src/pysweepme/Ports.py

```diff
@@ -830,6 +875,61 @@ def read_internal(self, digits=0):
         return answer
 
 
+class SOCKETport(Port):
+
+    def __init__(self, ID):
+
+        super(__class__, self).__init__(ID)
+
+    def open_internal(self):
+
+        port_ID = self.port_properties["ID"]
+        ok, HOST, PORT = is_IP(port_ID)
+        if not ok:
+            raise ValueError(f"Port string {[port_ID]} is not a valid IPV4 address")
+
+        self.port = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.port.settimeout(0.1)
+        self.port.connect((HOST, PORT))
+
+        # if self.port_properties["TCPIP_EOLwrite"] is not None:
+        #     self.port.write_termination = self.port_properties["TCPIP_EOLwrite"]
+
+        # if self.port_properties["TCPIP_EOLread"] is not None:
+        #     self.port.read_termination = self.port_properties["TCPIP_EOLread"]
+
+    def close_internal(self):
+        self.port.close()
+
+    def get_identification(self):
+
+        self.write("*IDN?")
+        return self.read()
+
+    def write_internal(self, cmd: str):
+
+        self.port.sendall(cmd.encode('latin-1'))
+        time.sleep(self.port_properties["delay"])
+
+    def read_internal(self, digits=0):
+
+        start_t = time.time()
+        received = False
+
+        while time.time() - start_t < float(self.port_properties["timeout"]):
+            try:
+                answer = self.port.recv(1024)
```

Comment:
read_internal comes with a digits argument that could be by default 1024 so that digits is used in self.port.recv(digits). This way people can change in the driver how many bytes need to be read by using there something like self.port.read(10)

Reply:
that's not that useful as I understand it because digits is the **max** readlength for socket.
More of an issue is what happens if the instrument sends more than 1024 bytes. There the code breaks.